velox/exec/LocalPlanner.cpp

```diff
@@ -411,6 +412,14 @@ std::shared_ptr<Driver> DriverFactory::createDriver(
           assignUniqueIdNode,
           assignUniqueIdNode->taskUniqueId(),
           assignUniqueIdNode->uniqueIdCounter()));
+    } else if (
+        auto markDistinctNode =
+            std::dynamic_pointer_cast<const core::MarkDistinctNode>(planNode)) {
+      std::vector<uint32_t> channels = toChannels(
```

Comment:
We can do this logic inside the constructor of the MarkDistinct operator. The distinct fields can be directly obtained from the PlanNode there, so there isn't a need for the extra variable for the input channels.